src/components/Analyze/AnalyzeContext.tsx

```diff
@@ -15,16 +17,41 @@ function calculateTotalPrice(consumptionData: analyzeDataType[]) {
 	return totalPrice
```

Comment:
배열의 각 요소를 순회하며 값을 누적시켜야 한다면, `reduce` 함수는 어떤가요?
map은 새 배열을 만들지만 현재 코드에서는 그것을 사용하지 않으므로 불필요한 메모리 사용이 있습니다.

`reduce`는 callback함수의 실행 값을 누적하여 하나의 결과값을 반환 합니다.
map은 새로운 배열을 반환하는데, 여기서는 새로운 배열을 하나의 결과값만을 반환하고, 코드 길이도 간결해집니다.
```suggestion
function calculateTotalPrice(consumptionData: analyzeDataType[]) {
  const totalPrice = consumptionData.reduce((total, each) => total + each.price, 0)
  return totalPrice
}
```
혹은 `forEach`를 사용할 수도 있겠네요.
```ts
function calculateTotalPrice(consumptionData: analyzeDataType[]) {
  let totalPrice = 0;
  consumptionData.forEach((each) => (totalPrice += each.price));
  return totalPrice;
}
```

Reply:
역할상 reduce가 더 적합하니, reduce 사용이 더 좋을 것 같습니다!
하지만, 해당 코드에서는 map함수를 통해 새로운 배열을 만들지만, 그 배열이 선언되는게 아니기 때문에 거의 바로 메모리에서 없어진다고 알고 있습니다. 그래서 메모리 사용량에서는 유의미한 차이가 있을 것 같지는 않다는 생각입니다.
map을 사용해서 한 눈에 이게 뭔지 파악이 힘들지 않을까 싶어서 함수로 따로 뺐던건데, 이렇게 되면 함수 선언도 필요 없을 것 같아서 바로 넣으면 될 것 같습니다!

```suggestion
const totalPrice = consumptionData.reduce(
		(total, each) => total + each.price,
		0,
	)
```